### PR TITLE
Some comments explaining some of the constants in the code.

### PR DESCRIPTION
**Pull request by gmaxwell**

### CODE REVIEW

**Comment by droark** [comment]
Thanks, Greg. I've been doing a review of libsecp256k1 in my free time. Comments like these and the one in pull request #205 are a huge help. :)

File: src/ecdsa_impl.h [hunk 1]
```diff
@@ -15,11 +15,28 @@
 #include "ecmult_gen.h"
 #include "ecdsa.h"
 
+/** Group order for secp256k1 defined as 'n' in "Standards for Efficient Cryptography" (SEC2) 2.7.1
```
Comment (sipa) [inline src/ecdsa_impl.h]: This prints p, not n.

**Comment by sipa** [comment]
There are several copies of p and n in different places of the code. Maybe it's useful to only explain them once, and just put references in other places ("see the definition of secp256k1_scalar_const_order in file X").

**Comment by sipa** [comment]
ACK

**Pull request merged**